### PR TITLE
Reduce arenamanager allocations

### PR DESCRIPTION
**Pull request by moredure**
Reduce memory allocation with same path behaiviour, clean dir once instead of doing it each time.

As an alternative we can do
```go
var b strings.Builder
a := strconv.Itoa(i)
b.Grow(len(dir) + 1 + len(a) + len(cArenaFilePrefix))
b.WriteString(dir)
b.WriteByte('/')
b.WriteString(a)
b.WriteString(cArenaFilePrefix)
```
but it's slightly slower

```go

func BenchmarkSpeeds(b *testing.B) {
	b.Run("compare speeds", func(b *testing.B) {
		var dir = "/home/user/datadir"
		var p = ""
		b.Run("builder", func(b *testing.B) {
			b.ReportAllocs()
			for i := 0; i < b.N; i++ {
				var b strings.Builder
                                a := strconv.Itoa(i)
				b.Grow(len(dir) + 1 + len(a) + len("_arena.dat"))
				b.WriteString(dir)
				b.WriteByte('/')
				b.WriteString(a)
				b.WriteString("_arena.dat")
				p = b.String()
			}
		})
		b.Run("slice", func(b *testing.B) {
			var filePath []byte
			b.ReportAllocs()
			for i := 0; i < b.N; i++ {
				filePath = append(filePath[:0], dir...)
				filePath = append(filePath, '/')
				filePath = strconv.AppendInt(filePath, int64(i), 10)
				filePath = append(filePath, "_arena.dat"...)
				p = string(filePath)
			}
		})
		b.Run("path", func(b *testing.B) {
			b.ReportAllocs()
			for i := 0; i < b.N; i++ {
				fileName := strconv.Itoa(i) + "_arena.dat"
				p = path.Join(dir, fileName)
			}
		})
		runtime.KeepAlive(p)
	})
}

```

```
cpu: Intel(R) Core(TM) i5-7600 CPU @ 3.50GHz
BenchmarkSpeeds
BenchmarkSpeeds/compare_speeds
BenchmarkSpeeds/compare_speeds/builder
BenchmarkSpeeds/compare_speeds/builder-4 	15500703	        76.11 ns/op	      55 B/op	       1 allocs/op
BenchmarkSpeeds/compare_speeds/slice
BenchmarkSpeeds/compare_speeds/slice-4   	23539419	        56.07 ns/op	      47 B/op	       1 allocs/op
BenchmarkSpeeds/compare_speeds/path
BenchmarkSpeeds/compare_speeds/path-4    	 5428278	       203.3 ns/op	     103 B/op	       2 allocs/op
```
@mangalaman93, @ashish-goswami  please take a look!

### CODE REVIEW

**Comment by moredure** [comment]
As an alternative we can do
```go
var b strings.Builder
a := strconv.Itoa(i)
b.Grow(len(dir) + 1 + len(a) + len(cArenaFilePrefix))
b.WriteString(dir)
b.WriteByte('/')
b.WriteString(a)
b.WriteString(cArenaFilePrefix)
				```

File: arenamanager.go
Comment (mangalaman93) [inline arenamanager.go:24]: Could we call it something other than `filePath`. This happens to be a pkg name in Go.

**Review by mangalaman93 — CHANGES_REQUESTED** [review]
This looks good. Just one comment, thanks for the benchmark. It would be fine to have the benchmark in the repo too. No big deal.

File: arenamanager.go
Comment (moredure) [reply arenamanager.go:24]: Indeed, how about fullPath?

**Review by mangalaman93 — APPROVED** [review]

**Comment by mangalaman93** [comment]
Waiting for build to pass before we can merge it. Could be related to #78.

**Pull request merged**
